packages/@aws-cdk/aws-stepfunctions-tasks/lib/apigateway/invoke.ts

```diff
@@ -0,0 +1,277 @@
+  readonly path?: string;
+
+  /**
+   * Query strings string to list of strings
+   * @default - No query parameters
+   */
+  readonly queryParameters?: sfn.TaskInput;
+
+  /**
+   * HTTP Request body
+   * @default - No requestBody
+   */
+  readonly requestBody?: sfn.TaskInput;
+  /**
+   * Authentication methods
+   *
+   * NO_AUTH: call the API direclty with no authorization method
+   *
+   * IAM_ROLE: Use the IAM role associated with the current state machine for authorization
+   *
+   * RESOURCE_POLICY: Use the resource policy of the API for authorization
+   *
+   * @default - NO_AUTH
+   */
+  readonly authType?: sfn.TaskInput;
+
+}
+/**
+ * Invoke an API endpoint as a Task
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-api-gateway.html
+ */
+export class ApiGatewayInvoke extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: ApiGatewayInvokeProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, ApiGatewayInvoke.SUPPORTED_INTEGRATION_PATTERNS);
+    const authType = this.props.authType ? this.props.authType.value : sfn.TaskInput.fromDataAt('$.AuthType').value;
+    if (authType === 'IAM_ROLE') {
+      const resource = props.apiEndpoint.split('.', 1)[0] + '/' + (props.stage ? props.stage + '/' : '$default/') + props.method + '/' + (props.path ?? '');
+
+      this.taskPolicies = [
+        new iam.PolicyStatement({
+          resources: [
+            Stack.of(this).formatArn({
+              service: 'execute-api',
+              resource: resource,
+            }),
+          ],
+          actions: ['execute-api:Invoke'],
+        }),
+      ];
+    } else if (authType === 'RESOURCE_POLICY') {
+      if (!sfn.FieldUtils.containsTaskToken(props.headers)) {
+        throw new Error('Task Token is required in `headers` Use JsonPath.taskToken to set the token.');
+      }
+      const resource = props.apiEndpoint.split('.', 1)[0] + '/' + (props.stage ? props.stage + '/' : '') + props.method + '/' + (props.path ? props.path + '/*' : '*');
+
+      this.taskPolicies = [
+        new iam.PolicyStatement({
+          resources: [
+            Stack.of(this).formatArn({
+              service: 'execute-api',
+              resource: resource,
+            }),
+          ],
+          actions: ['execute-api:Invoke'],
+          conditions: {
+            StringEquals: {
+              'aws:SourceArn': '*',
+            },
+          },
+        }),
+      ];
```

Comment:
there's some duplication in this control flow block from the if statement that we can simplify